example_client/cpp/src/resnet_client_benchmark.cpp

```diff
@@ -0,0 +1,561 @@
+            return false;
+        }
+
+        std::filebuf* pbuf = imageFile.rdbuf();
+        auto fileSize = pbuf->pubseekoff(0, std::ios::end, std::ios::in);
+
+        auto image = std::unique_ptr<char[]>(new char[fileSize]());
+
+        pbuf->pubseekpos(0, std::ios::in);
+        pbuf->sgetn(image.get(), fileSize);
+        imageFile.close();
+
+        entriesOut.emplace_back(BinaryData{std::move(image), fileSize, entry.expectedLabel});
+    }
+
+    return true;
+}
+
+bool readImagesCvMat(const std::vector<Entry>& entriesIn, std::vector<CvMatData>& entriesOut, const tensorflow::string& layout) {
+    entriesOut.clear();
+
+    for (const auto& entryIn : entriesIn) {
+        CvMatData entryOut;
+        entryOut.layout = layout;
+        entryOut.expectedLabel = entryIn.expectedLabel;
+        try {
+            entryOut.image = cv::imread(entryIn.imagePath);
+            if (entryOut.image.data == nullptr) {
+                return false;
+            }
+        } catch (cv::Exception& ex) {
+            return false;
+        }
+        entryOut.image.convertTo(entryOut.image, CV_32F);
+        cv::resize(entryOut.image, entryOut.image, cv::Size(224, 224));
+        if (layout == "nchw") {
+            entryOut.image = reorderMatToNchw(&entryOut.image);
+        }
+        entriesOut.emplace_back(entryOut);
+    }
+
+    return true;
+}
+
+// TODO: Implement wrap-around
+template <typename T>
+std::vector<T> selectEntries(const std::vector<T>& entries, tensorflow::int64 batchSize, tensorflow::int64 iteration) {
+    size_t startPoint = (iteration * batchSize) % entries.size();
+    if (batchSize > entries.size()) {
+        std::vector<T> selected;
+        while (selected.size() < batchSize) {
+            auto remainingBatches = batchSize - selected.size();
+            if (entries.size() - startPoint >= remainingBatches) {
+                selected.insert(selected.end(), entries.begin() + startPoint, entries.begin() + startPoint + remainingBatches);
+                break;
+            }
+            selected.insert(selected.end(), entries.begin() + startPoint, entries.end());
+            startPoint = 0;
+        }
+        return selected;
+    } else {
+        if (startPoint + batchSize > entries.size()) {
+            std::vector<T> selected;
+            selected.insert(selected.end(), entries.begin() + startPoint, entries.end());
+            selected.insert(selected.end(), entries.begin(), entries.begin() + (batchSize - (entries.size() - startPoint)));
+            return selected;
+        } else {
+            return std::vector<T>(entries.begin() + startPoint, entries.begin() + startPoint + batchSize);
+        }
+    }
+}
+
+std::vector<tensorflow::int64> argmax(const tensorflow::Tensor& tensor) {
+    const auto& shape = tensor.shape();
+    assert(shape.dims() == 2);
+    size_t batchSize = shape.dim_size(0);
+    size_t elements = shape.dim_size(1);
+    std::vector<tensorflow::int64> labels;
```

Comment:
Resize just after creation since you know the final size?

Reply:
done